src/libraries/Common/src/System/Obsoletions.cs

```diff
@@ -159,5 +159,11 @@ internal static class Obsoletions
 
         internal const string JsonSerializerOptionsAddContextMessage = "JsonSerializerOptions.AddContext is obsolete. To register a JsonSerializerContext, use either the TypeInfoResolver or TypeInfoResolverChain properties.";
         internal const string JsonSerializerOptionsAddContextDiagId = "SYSLIB0049";
+
+        internal const string LegacyFormatterMessage = "Legacy formatter-based serialization is obsolete and should not be used.";
```

Comment:
Is there "formatter-based serialization" that isn't "legacy"?  Consider dropping the "legacy" word here and below.